ca/client.go

```diff
@@ -425,16 +425,17 @@ func parseEndpoint(endpoint string) (*url.URL, error) {
 }
 
 // ProvisionerOption is the type of options passed to the Provisioner method.
-type ProvisionerOption func(o *provisionerOptions) error
+type ProvisionerOption func(o *ProvisionerOptions) error
 
-type provisionerOptions struct {
-	cursor string
-	limit  int
-	id     string
-	name   string
+// ProvisionerOptions stores options for the provisioner CRUD API.
+type ProvisionerOptions struct {
```

Comment:
If we're going to use the `With*` methods from bellow, there's no need to change the visibility of this. Exporting new things is most costly if we want to change the implementation because we need to take into account backward compatibility.

Reply:
I use the ca.ProvisionerOptions in the following places:
https://github.com/smallstep/cli/pull/673/files#diff-88a143d78812b71589b94b33a441ea5b91e1e52a2ab19d6c2306112e8fc9ef26R142
https://github.com/smallstep/cli/pull/673/files#diff-88a143d78812b71589b94b33a441ea5b91e1e52a2ab19d6c2306112e8fc9ef26R177
https://github.com/smallstep/cli/pull/673/files#diff-88a143d78812b71589b94b33a441ea5b91e1e52a2ab19d6c2306112e8fc9ef26R98

Reply:
Well ok, it can be done like that or creating its own options in the command. One not though, ID should have more precedence than Name.